crates/net/discv5/src/config.rs

```diff
@@ -0,0 +1,337 @@
+    /// Config used by [`discv5::Discv5`]. Contains the [`ListenConfig`], with the discovery listen
+    /// socket.
+    pub(super) discv5_config: discv5::Config,
+    /// Nodes to boot from.
+    pub(super) bootstrap_nodes: HashSet<BootNode>,
+    /// [`ForkId`] to set in local node record.
+    pub(super) fork: (&'static [u8], ForkId),
+    /// Mempool TCP port to advertise.
+    pub(super) tcp_port: u16,
+    /// Additional kv-pairs to include in local node record.
+    pub(super) other_enr_data: Vec<(&'static str, Bytes)>,
+    /// Allow no TCP port advertised by discovered nodes. Disallowed by default.
+    pub(super) allow_no_tcp_discovered_nodes: bool,
+    /// Interval in seconds at which to run a lookup up query with local node ID as target, to
+    /// populate kbuckets.
+    pub(super) self_lookup_interval: u64,
+    /// Optional filter rules to apply to a discovered peer in order to determine if it should be
+    /// passed up to mempool or dropped.
+    pub(super) filter_discovered_peer: Filter,
+}
+
+impl Config {
+    /// Returns a new [`ConfigBuilder`].
+    pub fn builder() -> ConfigBuilder {
+        ConfigBuilder::default()
+    }
+}
+
+impl<T> Config<T> {
+    /// Returns the discovery (UDP) socket contained in the [`discv5::Config`]. Returns the IPv6
+    /// socket, if both IPv4 and v6 are configured. This socket will be advertised to peers in the
+    /// local [`Enr`](discv5::enr::Enr).
+    pub fn discovery_socket(&self) -> SocketAddr {
+        match self.discv5_config.listen_config {
+            ListenConfig::Ipv4 { ip, port } => (ip, port).into(),
+            ListenConfig::Ipv6 { ip, port } => (ip, port).into(),
+            ListenConfig::DualStack { ipv6, ipv6_port, .. } => (ipv6, ipv6_port).into(),
+        }
+    }
+
+    /// Returns the mempool (TCP) socket contained in the [`discv5::Config`]. This socket will be
+    /// advertised to peers in the local [`Enr`](discv5::enr::Enr).
+    pub fn mempool_socket(&self) -> SocketAddr {
+        let port = self.tcp_port;
+        match self.discv5_config.listen_config {
+            ListenConfig::Ipv4 { ip, .. } => (ip, port).into(),
+            ListenConfig::Ipv6 { ip, .. } => (ip, port).into(),
+            ListenConfig::DualStack { ipv4, .. } => (ipv4, port).into(),
+        }
+    }
+}
+
+/// A boot node can be added either as a string in either 'enode' URL scheme or serialized from
+/// [`Enr`](discv5::Enr) type.
+#[derive(Debug, PartialEq, Eq, Hash)]
+pub enum BootNode {
+    /// An unsigned node record.
+    Enode(String),
+    /// A signed node record.
+    Enr(discv5::Enr),
+}
+
+/// Identifies a chain in an [`Enr`](discv5::enr::Enr). Used as key for the [`ForkId`] kv-pair in
+/// the [`Enr`](discv5::enr::Enr).
+pub trait IdentifyForkIdKVPair {
+    /// L1 EL
+    const ETH: &'static [u8] = b"eth";
+    /// L1 CL
+    const ETH2: &'static [u8] = b"eth2";
+    /// Optimism
+    const OPSTACK: &'static [u8] = b"opstack";
+}
```

Comment:
these should just be regular consts, with an enum. Given optimism has added an entry, I understand that we might want to abstract this in the future, but for now I'd prefer this to be simpler